### PR TITLE
More flexible partisans

### PR DESCRIPTION
**Pull request by lmoureaux**
This PR adds flexibility to partisans and changes the default behavior when they cost population:

* Multiple partisan unit types are now supported. The game will pick one the player can build, or the first one if the player can't build any
* When partisan units cost population, the population is taken away from the conquered city (rationale: it's an exodus)

## Testing

In the ruleset you're playing:

* Lower or remove requirements for the `Inspire_Partisans` effect
* Add the `Partisan` flag to some normal unit, e.g. Musketeers
* Change your favorite Partisan unit's `pop_cost`

Then conquer a city. You should see partisans of an appropriate unit type appear and the city size be reduced accordingly if they cost population. The city should never be wiped out.

### CODE REVIEW

File: server/unittools.cpp
Comment (blabber) [inline server/unittools.cpp:1181]: I guess this should read `// If the player can build any partisan, pick the first one`?

**Comment by blabber** [comment]
Deleted my comment after rereading the code. Sorry for the noise.

File: docs/Modding/Rulesets/effects.rst
Comment (jwrober) [inline docs/Modding/Rulesets/effects.rst:817]: Make this an RST note `.. note::`?

**Review by jwrober — CHANGES_REQUESTED** [review]
One small edit to docs. I do want to test before a full approval. Looks great so far.

**Review by jwrober — APPROVED** [review]
Other than the small docs change, rest looks great. Was able to follow steps in a test game.

File: docs/Modding/Rulesets/effects.rst
Comment (lmoureaux) [reply docs/Modding/Rulesets/effects.rst]: Updated:
image

**Review by jwrober — APPROVED** [review]

**Pull request merged**
